kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/implementations/persistentOrderedMap/PersistentOrderedMapBuilder.kt

```diff
@@ -93,14 +93,16 @@ internal class PersistentOrderedMapBuilder<K, V>(private var map: PersistentOrde
             val previousLinks = mapBuilder[links.previous]!!
 //            assert(previousLinks.next == key)
             mapBuilder[links.previous as K] = previousLinks.withNext(links.next)
+        } else {
+            firstKey = links.next as? K
```

Comment:
Why safe cast here? If `next`/`previous` is not that sentinel object then it's definitely of type `K`. Though the cast is unchecked anyway, replacing it with safe cast isn't a way to get rid of the warning.

Reply:
The if statement above checks only `previous`. `next` may be the sentinel object regardless of `previous`.

Reply:
Suppose map contained a single element, the links `previous` and `next` are both the sentinel object.

Reply:
Ok, but then this check won't work as expected given that `K` is erased to upper bound (`Object`) at run time

Reply:
Now that the `firstKey` and `lastKey` are assumed to hold either instance of `K` or `LinkEnd`, this safe cast is unnecessary.